google-cloud-firestore/src/main/java/com/google/cloud/firestore/VectorQuery.java

```diff
@@ -0,0 +1,191 @@
+   */
+  @Override
+  public ApiFuture<VectorQuerySnapshot> get() {
+    return get(null, null);
+  }
+
+  /**
+   * Plans and optionally executes this VectorQuery. Returns an ApiFuture that will be resolved with
+   * the planner information, statistics from the query execution (if any), and the query results
+   * (if any).
+   *
+   * @return An ApiFuture that will be resolved with the planner information, statistics from the
+   *     query execution (if any), and the query results (if any).
+   */
+  @Override
+  public ApiFuture<ExplainResults<VectorQuerySnapshot>> explain(ExplainOptions options) {
+    return super.explain(options);
+  }
+
+  /**
+   * Returns true if this VectorQuery is equal to the provided object.
+   *
+   * @param obj The object to compare against.
+   * @return Whether this VectorQuery is equal to the provided object.
+   */
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || !(obj instanceof VectorQuery)) {
+      return false;
+    }
+    VectorQuery otherQuery = (VectorQuery) obj;
+    return Objects.equals(query, otherQuery.query)
+        && Objects.equals(vectorField, otherQuery.vectorField)
+        && Objects.equals(queryVector, otherQuery.queryVector)
+        && Objects.equals(options, otherQuery.options)
+        && limit == otherQuery.limit
+        && distanceMeasure == otherQuery.distanceMeasure;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(query, options, vectorField, queryVector, limit, distanceMeasure);
+  }
+
+  @Override
+  protected RunQueryRequest.Builder toRunQueryRequestBuilder(
+      @Nullable final ByteString transactionId,
+      @Nullable final Timestamp readTime,
+      @Nullable ExplainOptions explainOptions) {
+
+    // Builder for the base query
+    RunQueryRequest.Builder requestBuilder =
+        query.toRunQueryRequestBuilder(transactionId, readTime, explainOptions);
+
+    // Builder for find nearest
+    StructuredQuery.FindNearest.Builder findNearestBuilder =
+        requestBuilder.getStructuredQueryBuilder().getFindNearestBuilder();
+    findNearestBuilder.getQueryVectorBuilder().setMapValue(this.queryVector.toProto());
+    findNearestBuilder.getLimitBuilder().setValue(this.limit);
+    switch (this.distanceMeasure) {
+      case COSINE:
+        findNearestBuilder.setDistanceMeasure(StructuredQuery.FindNearest.DistanceMeasure.COSINE);
+        break;
+      case EUCLIDEAN:
+        findNearestBuilder.setDistanceMeasure(
+            StructuredQuery.FindNearest.DistanceMeasure.EUCLIDEAN);
+        break;
+      case DOT_PRODUCT:
+        findNearestBuilder.setDistanceMeasure(
+            StructuredQuery.FindNearest.DistanceMeasure.DOT_PRODUCT);
+        break;
+      default:
+        findNearestBuilder.setDistanceMeasure(
+            StructuredQuery.FindNearest.DistanceMeasure.UNRECOGNIZED);
+    }
```

Comment:
I think a helper method could cleanup this code, and remove some repetition:

```
  private static StructuredQuery.FindNearest.DistanceMeasure toProto(DistanceMeasure distanceMeasure) {
    switch (distanceMeasure) {
      case COSINE:
        return StructuredQuery.FindNearest.DistanceMeasure.COSINE;
      case EUCLIDEAN:
        return StructuredQuery.FindNearest.DistanceMeasure.EUCLIDEAN;
      case DOT_PRODUCT:
        return StructuredQuery.FindNearest.DistanceMeasure.DOT_PRODUCT;
      default:
        return StructuredQuery.FindNearest.DistanceMeasure.UNRECOGNIZED;
    }
  }
```

Reply:
Yes! Very nice. Thanks.